docs/adrs/003-bugzilla-response-codes.md

```diff
@@ -0,0 +1,185 @@
+- [743 - Create alerts for when bugs fail to sync](https://github.com/mozilla/jira-bugzilla-integration/issues/743)
+
+## Context and Problem Statement
+When Bugzilla receives too many error responses from JBI, it stops triggering webhook calls for the entire project causing data to stop syncing. Frequently, these errors are due to a configuration error in Jira or Bugzilla. JBI being unable to process a payload due to errors in configuration (or incomplete configuration) in Jira or a mismatch of data for a single bug. These outages can last multiple days in some cases.
+
+We don't want the entire sync process to stop because of this. We have identified four options to solve this problem.
+
+## Decision Drivers
+- Amount of initial engineering effort
+- Amount of maintenance effort
+- Overall performance of JBI (how quickly is data able to move)
+- How intuitive the solution is to the users that depend on the data (will picking the easiest option solve their needs?)
+
+## Proposed Solution
+We propose to use a data bucket as DLQ. Events can attempt to be reprocessed from here, but we will skip any events identified as stale. See the diagram below:
+
+![Flow chart detailing the data flow, see expandable below for full details](./003.drawio.jpg "Proposed Solution Flow Chart")
+
+<details>
+
+  <summary>Breakdown of flowchart</summary>
+
+  1. JBI receives a payload from Bugzilla or the Retry Scheduler.
+  1. JBI will always return 200/OK for a response.
+  1. If the bug is private, discard the event and log why.
+  1. If the bug cannot be found in the Bugzilla API, discard the event and log why.
+  1. If an associated action cannot be found for the event, discard the event and log why.
+  1. If a matching Jira issue cannot be found, and the event is not creating one, discard the event and log why.
+  1. If there is a mismatch between project keys in the event and Jira, discard the event and log why.
+  1. If there is a blocking event in the DLQ, skip to the Error Event Handler.
+  1. If there is a merge-able event in the DLQ, merge with the current event (current event wins conflicts).
+  1. Write updated data to Jira's API.
+      1. If successful, delete any associated items in DLQ.
+      1. If error is returned, continue to Error Event Handler
+  1. Handle errors in Error Event Handler
+      1. Write an error to the logs, which may be forwarded to an alerting mechanism.
+      1. Write an updated event file to the DLQ if the original event is less than 7 days old.
+      1. If we have exceeded 7 days from the original event, delete associated DLQ items.
+  1. The retry scheduler runs every 12 hours and will re-send events to JBI. Starting with potentially blocking events first.
+</details>
+
+### Pros:
+ - Avoids the problem of accidentally overwriting newer data with older data
+ - Avoids making users correct data manually if something is misconfigured
```

Comment:
I didn't get that one 🤔 
How does it prevent users from correcting? Correcting what? Fields on Jira?

Reply:
Sorry, that's compared to Option 1 which is "log the failure and drop the data". In that scenario, users would be expected to manually correct any data that failed to sync unless they wanted to force a re-sync somehow from Bugzilla.